front-end/src/screens/Settings/address.tsx

```diff
@@ -145,53 +150,74 @@ const Address = ({ className }: Props): JSX.Element => {
 	const linkIcon = <><Icon name='chain'/>Link</>;
 	const unlinkIcon = <><Icon name='broken chain'/>Unlink</>;
 
-	// TODO: generate avtar image svg logo
+	if (extensionNotAvailable) {
+		return (
+			<Form className={className} standalone='false'>
+				<Form.Group>
+					<Form.Field width={16}>
+						<div className='text-muted'>No address detected.</div>&nbsp;
+						<div className='text-muted'>Please reload this page with <a href={getExtensionUrl()}>Polkadot extension</a>.</div>
+					</Form.Field>
+				</Form.Group>
+			</Form>
+		);
+	}
+
+	if (accounts.length === 0) {
+		return (
+			<Form className={className} standalone='false'>
+				<Form.Group>
+					<Form.Field width={16}>
+						<label>Linked address</label>
+						<div className='text-muted'>Associate your account with an on chain address using the Polkadot-js extention.</div>&nbsp;
+						<div className='link-button-container'>
+							<Button primary onClick={handleDetect}>
+								Link Address
```

Comment:
The text should be "Show linked addresses" if the user has linked addresses already (available from the context).

Reply:
done